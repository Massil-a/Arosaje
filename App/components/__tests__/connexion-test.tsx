import React from 'react';
import { render } from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';
import ConnexionScreen from '../../app/Login/connexion';
import AsyncStorage from '@react-native-async-storage/async-storage';

jest.mock('@react-native-async-storage/async-storage', () => ({
    setItem: jest.fn(),
    getItem: jest.fn(),
    removeItem: jest.fn(),
    clear: jest.fn(),
}));

describe('ConnexionScreen', () => {
  it('renders correctly', () => {
    const { toJSON } = render(
      <NavigationContainer>
        <ConnexionScreen setIsModalVisible={() => {}} />
      </NavigationContainer>
    );

    // Générer le snapshot
    expect(toJSON()).toMatchSnapshot();
  });
});
