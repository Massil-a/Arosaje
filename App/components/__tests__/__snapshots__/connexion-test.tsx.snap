// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnexionScreen renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#FFF",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#668F80",
        "flexDirection": "row",
        "height": 100,
        "justifyContent": "space-between",
        "paddingHorizontal": 10,
        "paddingTop": 20,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 50,
        }
      }
    >
      <Text
        style={
          {
            "color": "#FFF",
            "fontSize": 24,
            "fontWeight": "bold",
          }
        }
      >
        Connexion
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "flex-start",
        "marginTop": 30,
      }
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Email"
      placeholderTextColor="#BDBDBD"
      style={
        {
          "alignSelf": "center",
          "backgroundColor": "#F6F6F6",
          "borderRadius": 5,
          "height": 50,
          "margin": 10,
          "padding": 10,
          "width": "90%",
        }
      }
      value="a@b.com"
    />
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <TextInput
        autoCapitalize="words"
        onChangeText={[Function]}
        placeholder="Mot de passe"
        placeholderTextColor="#BDBDBD"
        secureTextEntry={true}
        style={
          {
            "backgroundColor": "#F6F6F6",
            "borderRadius": 5,
            "height": 50,
            "margin": 10,
            "padding": 10,
            "width": "90%",
          }
        }
        value="Azerty12345!"
      />
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "position": "absolute",
            "right": 30,
          }
        }
      >
        <Text />
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginTop": 10,
          "width": "100%",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 32,
            "justifyContent": "center",
            "width": 32,
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="100%"
          bbWidth="100%"
          focusable={false}
          meetOrSlice={0}
          minX={-10}
          minY={-10}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "flex": 0,
                "height": "100%",
                "width": "100%",
              },
            ]
          }
          vbHeight={84}
          vbWidth={84}
        >
          <RNSVGGroup
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
          >
            <RNSVGDefs>
              <RNSVGClipPath
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
                name="clipPath"
              >
                <RNSVGPath
                  d="M24 0.5H40C48.5809 0.5 54.4147 2.18067 58.117 5.88299C61.8193 9.58532 63.5 15.4191 63.5 24V40C63.5 48.5809 61.8193 54.4147 58.117 58.117C54.4147 61.8193 48.5809 63.5 40 63.5H24C15.4191 63.5 9.58532 61.8193 5.88299 58.117C2.18067 54.4147 0.5 48.5809 0.5 40V24C0.5 15.4191 2.18067 9.58532 5.88299 5.88299C9.58532 2.18067 15.4191 0.5 24 0.5Z"
                  fill={
                    {
                      "payload": 4294967295,
                      "type": 0,
                    }
                  }
                  propList={
                    [
                      "fill",
                      "stroke",
                      "strokeLinecap",
                      "strokeLinejoin",
                    ]
                  }
                  stroke={
                    {
                      "payload": 4286611584,
                      "type": 0,
                    }
                  }
                  strokeLinecap={1}
                  strokeLinejoin={1}
                />
              </RNSVGClipPath>
            </RNSVGDefs>
            <RNSVGPath
              d="M15 31.1977C23.1081 36.4884 29.5946 43 29.5946 43C29.5946 43 37.5 25.5 69 1.5"
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
              onLayout={[Function]}
              propList={
                [
                  "stroke",
                  "strokeWidth",
                  "strokeOpacity",
                  "strokeDasharray",
                  "strokeDashoffset",
                  "strokeLinecap",
                  "strokeLinejoin",
                ]
              }
              stroke={
                {
                  "payload": 4284911488,
                  "type": 0,
                }
              }
              strokeDasharray={null}
              strokeDashoffset={null}
              strokeLinecap={1}
              strokeLinejoin={1}
              strokeOpacity={0}
              strokeWidth={10}
            />
            <RNSVGPath
              d="M24 0.5H40C48.5809 0.5 54.4147 2.18067 58.117 5.88299C61.8193 9.58532 63.5 15.4191 63.5 24V40C63.5 48.5809 61.8193 54.4147 58.117 58.117C54.4147 61.8193 48.5809 63.5 40 63.5H24C15.4191 63.5 9.58532 61.8193 5.88299 58.117C2.18067 54.4147 0.5 48.5809 0.5 40V24C0.5 15.4191 2.18067 9.58532 5.88299 5.88299C9.58532 2.18067 15.4191 0.5 24 0.5Z"
              fill={
                {
                  "payload": 0,
                  "type": 0,
                }
              }
              propList={
                [
                  "fill",
                  "stroke",
                  "strokeWidth",
                  "strokeLinecap",
                  "strokeLinejoin",
                ]
              }
              stroke={
                {
                  "payload": 4284911488,
                  "type": 0,
                }
              }
              strokeLinecap={1}
              strokeLinejoin={1}
              strokeWidth={7}
            />
            <RNSVGGroup
              clipPath="clipPath"
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
            >
              <RNSVGPath
                d="M15 31.1977C23.1081 36.4884 29.5946 43 29.5946 43C29.5946 43 37.5 25.5 69 1.5"
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
                onLayout={[Function]}
                propList={
                  [
                    "stroke",
                    "strokeWidth",
                    "strokeOpacity",
                    "strokeDasharray",
                    "strokeDashoffset",
                    "strokeLinecap",
                    "strokeLinejoin",
                  ]
                }
                stroke={
                  {
                    "payload": 4294967295,
                    "type": 0,
                  }
                }
                strokeDasharray={null}
                strokeDashoffset={null}
                strokeLinecap={1}
                strokeLinejoin={1}
                strokeOpacity={0}
                strokeWidth={10}
              />
            </RNSVGGroup>
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
      <Text
        style={
          {
            "fontSize": 14,
            "paddingLeft": 5,
          }
        }
      >
        Se souvenir de moi
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "marginTop": 100,
        "paddingTop": 100,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#E0E0E0",
          "borderRadius": 25,
          "flexDirection": "row",
          "marginBottom": 20,
          "overflow": "hidden",
          "width": "90%",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#668F80",
            "flex": 1,
            "justifyContent": "center",
            "paddingVertical": 20,
          }
        }
      >
        <Text
          style={
            {
              "color": "#FFF",
              "fontSize": 18,
              "fontWeight": "bold",
            }
          }
        >
          Connexion
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <Text
        style={
          {
            "color": "#668F80",
            "fontSize": 14,
            "fontWeight": "bold",
            "padding": 15,
          }
        }
      >
        Mot de passe oublié ?
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <Text
        style={
          {
            "color": "#668F80",
            "fontSize": 14,
            "fontWeight": "bold",
            "padding": 15,
          }
        }
      >
        Créer un compte
      </Text>
    </View>
  </View>
</View>
`;
